Setup of Travis CI and eslint complete

michaelvincerra opened the pull request:
https://trello.com/c/uNJYEVun/11-travis-ci

michaelvincerra committed:
Integrate Travis CI.

michaelvincerra committed:
Integrated eslint.

psbanka commented on package.json:
```diff
@@ -10,7 +10,7 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
+    "test": "eslint --fix src && react-scripts test --env=jsdom",
```
better not to have the `test` command run `--fix`

michaelvincerra replied:
Removed.

psbanka commented on src/App.js:
```diff
@@ -1,32 +1,29 @@
+  constructor (props) {
+    super(props)
+    this.state = {
+      data: []
```
This state items does not appear to be used? If it's not used yet, let's not put it in. Don't write code you don't have to write! 😄

michaelvincerra replied:
OK, I removed it. Yet I'm not clear on the constructor. After talking with Alex last night, I understood that this (data: []) would be used to bind data that would later be called in ComponentWillMount. Maybe you can clarify this for me--aside from the PR.

psbanka commented on src/ChartVis.js:
```diff
@@ -2,66 +2,62 @@ import React, { Component } from 'react'
-    }
-
-    render() {
+  // 30.11.2017. Restart here.
```
Useless comment?

michaelvincerra replied:
Yes. Removed.

psbanka commented on src/ChartVis.js:
```diff
@@ -2,66 +2,62 @@ import React, { Component } from 'react'
+            'year': datapoint.date,
+            'value': datapoint.value
+          }
+        })
```
Nice use of `map()` !

psbanka commented on src/ChartVis.js:
```diff
@@ -2,66 +2,62 @@ import React, { Component } from 'react'
+      .catch(error => console.log(error))
+  }
+
+  componentDidMount () {
```
Let's don't write empty methods like this, `d3Function()`, and `italyGDPChart()`. These really serve no purpose and clutter the code. All the code should have a purpose, and nothing else should be left in.

michaelvincerra replied:
Will remove them.

psbanka commented on src/ChartVis.js:
```diff
@@ -2,66 +2,62 @@ import React, { Component } from 'react'
+  }
+
+  render () {
+    // call js functions here
```
If this were a professional situation, I would tell you to pull this comment and the one on line 50. However, if these help you get started, then go ahead and leave them in. 😄

michaelvincerra replied:
Yes, this helps me. But I will remember not to do this in a  professional situation.

psbanka commented on src/ChartVis.js:
```diff
@@ -2,66 +2,62 @@ import React, { Component } from 'react'
+    console.log(this.state.data)
+    return (
+    // chart id is for SVG
+      <section className='all_container'>
```
What is the purpose of assigning `className` to this DOM element? And the one on line 53 below?

michaelvincerra replied:
I'm referencing the CSS.

michaelvincerra committed:
Checkpoint prior to downgrading React.

michaelvincerra committed:
Revised ChartVis based on the RD3 library from https://www.npmjs.com/package/rd3. Simplified lifecycle hook functions. Revised fetchData() so that filteredData matches the data object structure with that of World Bank API.

michaelvincerra committed:
Refactored AnalistApp using RD3 and cleaned repo

updated dependencies for d3
Added  Roboto and Exo 2 fonts.
Added changes to styles
In App.js, reformatted with eslint
Cleaned up unused BarChart.js

Revised ChartVis based on the RD3 library from https://www.npmjs.com/package/rd3. Simplified lifecycle hook functions. Revised fetchData() so that filteredData matches the data object structure with that of World Bank API. Files uploading.

michaelvincerra committed:
Add jest. jest-fetch-mock and react-test-renderer.

cleanup App.js.

App.test.js: removes cruft tests.

ChartVis.js cleanup. Simplifies fetchData, componentWillMount, and adds parseData function.

michaelvincerra committed:
Completes one test for src/ChartVis.js and adds console.log() to ChartVis.js to inspect data.

michaelvincerra committed:
Adds padding to chart for proper view.

michaelvincerra committed:
Removes unused code.

michaelvincerra committed:
Removes unused tests.

michaelvincerra committed:
Removes test.

michaelvincerra committed:
Removes test.

michaelvincerra commented:
Hello Peter, 
I'm requesting your approval on this pull request. Will you please indicate if I've done it correctly? Thanks.

psbanka commented:
Looks good, @michaelvincerra !
